### ISSUE TITLE
Bug: 8.0.0.alpha24 shared configuration with type checking fail to start

### ISSUE THREAD

**Issue by jerome-benoit** [typescript-eslint/typescript-eslint#9185]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.5&fileType=.tsx&code=FBA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkeAhpgObp4VQCiAygEwAMPHpAA04GpHiwAJgFcM7SADlp6AIwA2EWNpT00VI3oFi7PJhnpRNKLv2HjRACLJMCgHQB6VMgBGHqXhkVuIkZsjQbJRmFsG0AG5YPkbI8ACy0nLoXACeRAwAHqbmltpQAGawmNDoAMLEJMih6Hk19A1ETACSRABiaOiK9PDoZFHFYgC%2BwZDIRPoyuuwA2qWQAFRrHmtugZBiALrT6Pnzi5Qr1pBEKgD6krIYQavePnsQ%2ByATQA&tokens=false

### Repro Code

```TypeScript
Not relevant
```

### ESLint Config

```javascript
import simpleImportSort from 'eslint-plugin-simple-import-sort'
import tseslint from 'typescript-eslint'

export default defineFlatConfig([
  ...tseslint.config(...tseslint.configs.strictTypeChecked, ...tseslint.configs.stylisticTypeChecked),
  {
    plugins: {
      'simple-import-sort': simpleImportSort
    },
    rules: {
      'simple-import-sort/imports': 'error',
      'simple-import-sort/exports': 'error',
    }
  }
])
```

### tsconfig

```jsonc
{
  "$schema": "https://json.schemastore.org/tsconfig",
  "compilerOptions": {
    "target": "ES2022",
    "module": "Node16",
    "declaration": true,
    "declarationDir": "./lib/dts",
    "strict": true,
    "verbatimModuleSyntax": true,
    "forceConsistentCasingInFileNames": true
  },
  "include": ["**/*.ts"],
  "exclude": ["node_modules", "lib"]
}
```

### Expected Result

No error at running eslint shared configuration with type checking. 

### Actual Result

`Error: Error while loading rule '@typescript-eslint/await-thenable': You have used a rule which requires parserServices to be generated. You must therefore provide a value for the "parserOptions.project" property for @typescript-eslint/parser. Parser: undefined Note: detected a parser other than @typescript-eslint/parser. Make sure the parser is configured to forward "parserOptions.project" to @typescript-eslint/parser.`

**Comment by bradzacher** [comment]
This is working as expected. Have you read our getting started guide?
https://typescript-eslint.io/getting-started/typed-linting
